MetaMask/metamask-extension
# Approval flow for add & switch network

## Explanation



- Adopts the new `ApprovalController` version with the `startFlow`, `endFlow` and `setFlowLoadingText` methods.
- Adapts the add and switch network feature to use an approval flow without loading text.

The PR itself should not change any existing functionality. It just wraps the add & switch network feature with the new approval flow, which should prevent the notification window from closing in between approvals.

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/568
Fixes: https://github.com/MetaMask/MetaMask-planning/issues/569
Fixes: https://github.com/MetaMask/MetaMask-planning/issues/571

## Manual Testing Steps



- Go to the `Ethereum Chain Interactions` section on the Test Dapp
- Boot your local ganache on port 8546 and chainId 1337
- Click `ADD LOCALHOST 8546`. It should first open to add the chain and then switch to it.

OR

- Go to https://chainlist.org/
- Test adding and switching to any network

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
